processor/k8sattributesprocessor/processor.go

```diff
@@ -151,46 +151,60 @@ func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pco
 // addContainerAttributes looks if pod has any container identifiers and adds additional container attributes
 func (kp *kubernetesprocessor) addContainerAttributes(attrs pcommon.Map, pod *kube.Pod) {
 	containerName := stringAttributeFromMap(attrs, conventions.AttributeK8SContainerName)
-	if containerName == "" {
+	containerID := stringAttributeFromMap(attrs, conventions.AttributeContainerID)
+	var (
+		containerSpec *kube.Container
+		ok            bool
+	)
+	switch {
+	case containerName != "":
+		containerSpec, ok = pod.Containers.ByName[containerName]
+		if !ok {
+			return
+		}
+	case containerID != "":
+		containerSpec, ok = pod.Containers.ByID[containerID]
+		if !ok {
+			return
+		}
+	default:
 		return
 	}
-	containerSpec, ok := pod.Containers[containerName]
-	if !ok {
-		return
+	if _, ok := attrs.Get(conventions.AttributeK8SContainerName); !ok && containerSpec.Name != "" {
+		attrs.PutStr(conventions.AttributeK8SContainerName, containerSpec.Name)
 	}
-
-	if containerSpec.ImageName != "" {
-		if _, found := attrs.Get(conventions.AttributeContainerImageName); !found {
-			attrs.PutStr(conventions.AttributeContainerImageName, containerSpec.ImageName)
-		}
+	if _, ok := attrs.Get(conventions.AttributeContainerImageName); !ok && containerSpec.ImageName != "" {
+		attrs.PutStr(conventions.AttributeContainerImageName, containerSpec.ImageName)
 	}
-	if containerSpec.ImageTag != "" {
-		if _, found := attrs.Get(conventions.AttributeContainerImageTag); !found {
-			attrs.PutStr(conventions.AttributeContainerImageTag, containerSpec.ImageTag)
-		}
+	if _, ok := attrs.Get(conventions.AttributeContainerImageTag); !ok && containerSpec.ImageTag != "" {
+		attrs.PutStr(conventions.AttributeContainerImageTag, containerSpec.ImageTag)
 	}
-
-	runID := -1
-	runIDAttr, ok := attrs.Get(conventions.AttributeK8SContainerRestartCount)
-	if ok {
-		containerRunID, err := intFromAttribute(runIDAttr)
-		if err != nil {
-			kp.logger.Debug(err.Error())
-		} else {
-			runID = containerRunID
-		}
+	if _, ok := attrs.Get(conventions.AttributeContainerID); !ok && containerSpec.ID != "" {
+		attrs.PutStr(conventions.AttributeContainerID, containerSpec.ID)
```

Comment:
`ID` cannot be part of `containerSpec`. Every container status has its own ID. Looks like here we will always be setting `conventions.AttributeContainerID` attribute to the latest container status ignoring provided `conventions.AttributeK8SContainerRestartCount`

Reply:
I see what you mean. Let me revert this change back to what it was.